processor/spanmetricsprocessor/processor_test.go

```diff
@@ -282,6 +350,90 @@ func TestProcessorConsumeTraces(t *testing.T) {
 	}
 }
 
+func TestResourceCopying(t *testing.T) {
+	// Prepare
+	mexp := &mocks.MetricsExporter{}
+	tcon := &mocks.TracesConsumer{}
+
+	mexp.On("ConsumeMetrics", mock.Anything, mock.MatchedBy(func(input pdata.Metrics) bool {
+		rm := input.ResourceMetrics()
+		require.Equal(t, 2, rm.Len())
+
+		var rmA, rmB pdata.ResourceMetrics
+		rm0 := rm.At(0)
+		rm1 := rm.At(1)
+		serviceName, ok := getServiceName(rm0)
+		assert.True(t, ok, "should get service name from resourceMetric")
+
+		if serviceName == "service-a" {
+			rmA = rm0
+			rmB = rm1
+		} else {
+			rmB = rm0
+			rmA = rm1
+		}
+
+		require.Equal(t, 4, rmA.Resource().Attributes().Len())
+		require.Equal(t, 4, rmA.InstrumentationLibraryMetrics().At(0).Metrics().Len())
+		require.Equal(t, 2, rmB.Resource().Attributes().Len())
+		require.Equal(t, 2, rmB.InstrumentationLibraryMetrics().At(0).Metrics().Len())
+
+		wantResourceAttrServiceA := map[string]string{
+			resourceAttr1:          "1",
+			resourceAttr2:          "2",
+			notInSpanResourceAttr0: defaultNotInSpanAttrVal,
+			serviceNameKey:         "service-a",
+		}
+		rmA.Resource().Attributes().Range(func(k string, v pdata.AttributeValue) bool {
+			value := v.StringVal()
+			switch k {
+			case notInSpanResourceAttr1:
+				assert.Fail(t, notInSpanResourceAttr1+" should not be in this metric")
+			default:
+				assert.Equal(t, wantResourceAttrServiceA[k], value)
+				delete(wantResourceAttrServiceA, k)
```

Comment:
Why not use `assert.Equal` here? 

It could be like:

```golang
assert.NotEqual(t, k, notInSpanResource, "Must not be resource defined...")
assert.Equal(t, wantResourceAttrService[k], v.StringVal())
delete(wantResourceAttrServiceA, v.StringVal())
```